web/server/routers/post.ts

```diff
@@ -126,62 +134,53 @@ export const postRouter = router({
       z.object({
         email: z.string(),
         postOid: zodOidType,
-    })
-  ).mutation(async({input}) => {
-    const user = await findUserByEmail(input.email);
-    if (!user) {
-      throw new TRPCError({
-        message: "No user with given email exists.",
-        code: "BAD_REQUEST",
-      });
-    }
-    let email = "";
-    try {
-      const post = await getPost(input.postOid);
-      switch(post.type) {
-        case "fosterMove":
-          email = "foster@angelsrescue.org";
-          break;
-        case "return":
-          email = "returns@angelsrescue.org, foster@angelsrescue.org";
-          break;
-        case "temporary":
-          email = "tempfoster@angelsrescue.org";
-          break;
-        case "boarding":
-          email = "boardingadmin@angelsrescue.org"
-        case "shelter":
-        case "ownerSurrender":
-          email = "fosteroffer@angelsrescue.org";
-          break;
-      }
-      if (!email) {
+      })
+    )
+    .mutation(async ({ input }) => {
+      try {
+        const user = await findUserByEmail(input.email);
+        if (!user) {
+          throw new TRPCError({
+            message: "No user with given email exists.",
+            code: "BAD_REQUEST",
+          });
+        }
+      } catch (e) {
         throw new TRPCError({
-          message: "Post has invalid type.",
-          code: "BAD_REQUEST",
+          message: "Server unable to access user endpoint.",
+          code: "INTERNAL_SERVER_ERROR",
         });
       }
       try {
-        const info = await transporter.sendMail({
-          from: '"Angels Among Us Fostering Portal" <bitsofgood.aau@gmail.com>', 
-          to: email, 
-          subject: "Someone is ready to foster your dog!", 
-          text: "User has signed up to foster dog, a stray dog.", 
-        });
+        const post = await getPost(input.postOid);
+        const email = (fosterTypeEmails as any)[post.type] || "";
```

Comment:
Avoid type assertions on entire records if possible. The type of `fosterTypeEmails` will never change, so use a type assertion on the indexing variable instead (`post.type`).
```js
const email = fosterTypeEmails[post.type as FosterType]
```
You could also strongly type the return value of `getPost` if you don't want to use a type assertion.